SPA web security intro - post 1 (#1203)
oktadeveloper/okta-blog: _source/_posts/2022-06-30-spa-web-security.md

---
layout: blog_post
title: "Defense Against the Dark Hacks for SPAs"
author: alisa-duncan
by: advocate
communities: [security,javascript]
description: ""
tags: [security, react, vue, angular]
tweets:
- ""
- ""
- ""
image:
type: awareness
---

> There's a lot of information floating out there about web security. But when I read through the material, I noticed some information wasn't up to date, or they were written specifically for traditional server-rendered web applications, or they recommended anti-patterns. In a series of posts, I will cover web security concerns that all web devs should be aware of, emphasizing client-side applications, namely Single Page Applications (SPA). Furthermore, I'm not going to get into the nitty-gritty of web security, cryptography, and networking, and instead, focus on the main takeaways and high-level to-do lists. I'll also provide links where you can dive in deeper.

Reviewer: ```suggestion
There's a lot of information floating out there about web security. But when I read through the material, I noticed some information wasn't up to date, or it was written specifically for traditional server-rendered web applications, or the material recommended anti-patterns. In a series of posts, I will cover web security concerns that all web devs should be aware of, emphasizing client-side applications, namely Single Page Applications (SPA). Furthermore, I'm not going to get into the nitty-gritty of web security, cryptography, and networking, and instead, focus on the main takeaways and high-level to-do lists. I'll also provide links where you can dive in deeper.
```
keep it singular (the information)

<!-- Table with all the posts here as each post publishes -->

> So why do we worry about web security anyway? When we have vulnerabilities in our applications, it's an avenue for exploitation. In the wrong hands, exploiting the vulnerabilities can cause risks to your application, data, reputation, and bottom line. In the words of my coworker [Aaron Parecki](https://developer.okta.com/blog/authors/aaron-parecki/), it all boils down to one thing—liability. And we don't want to expose ourselves, our applications, and our companies to liability if we can help it. If you feel threatened, it's a good time to read on and identify how to harden your application.

Reviewer: would it be redundant to say "exploitation by bad actors" ?

Author: @havi oops, i missed this one earlier. Do you mean

> So why do we worry about web security anyway? When we have vulnerabilities in our applications, it's an avenue for exploitation by bad actors. In the wrong hands...

I don't think it's redundant. I'll make the change
